addons/sale_project/models/sale_order.py

```diff
@@ -26,6 +26,8 @@ class SaleOrder(models.Model):
     show_create_project_button = fields.Boolean(compute='_compute_show_project_and_task_button', groups='project.group_project_user')
     show_project_button = fields.Boolean(compute='_compute_show_project_and_task_button', groups='project.group_project_user')
     show_task_button = fields.Boolean(compute='_compute_show_project_and_task_button', groups='project.group_project_user')
+    closed_task_count = fields.Integer(compute='_compute_tasks_ids', string="Completed tasks associated with this sale")
+    completed_task_percentage = fields.Float(compute="_compute_completed_task_percentage", string="Completed tasks precentage associated with this sale")
```

Comment:
Don't you think it is possible to only have `completed_task_percentage` field (no `closed_task_count`),
and to compute it directly in `_compute_completed_task_percentage`?

Reply:
Also we need closed task count or open task count is needed for percentage calculation. `closed_task_count` is essential field to be displayed in the UI side as the readability on the stat button should be as `closed tasks / total_tasks (completion_percentage)`. Also  `closed_task_count` is the same field we are getting in the new read_group(i.e., the one you said to be made into one)
![image](https://github.com/odoo/odoo/assets/153157625/d7edee55-896b-4f60-bc2a-6b15ebfdfad0)

Reply:
Indeed. Sorry I didn't think about that. Let's keep the field, but compute both in the same method! :+1: